tuareg.el

```diff
@@ -489,6 +495,19 @@ Valid names are `browse-url', `browse-url-firefox', etc."
   "Return non-nil if point is inside an OCaml literal or comment."
   (nth 8 (syntax-ppss pos)))
 
+(defun tuareg--point-after-comment-p ()
+  "Return non-nil if a comment (possibly inside another one)
+precedes the point."
+  (and (eq (char-before) ?\))
+       (eq (char-before (1- (point))) ?*)
+       (save-excursion
+         (let ((pt (point)))
+           ;; A solution based on a single call to `syntax-ppss'
+           ;; takes > 1.5 the time of the following one.
+           (forward-comment -1)
+           (forward-comment 1)
+           (eq pt (point))))))
```

Comment:
There is a missing `and` inside the `let` here, without which the return values of the calls to `forward-comment` are ignored and that cannot have been the intent. However, I think Stefan is right in that the second `forward-comment` really isn't necessary: I cannot think of a case where it (and the subsequent comparison) would make a difference.

That said, this algorithm gives a false positive for comments inside string literals, such as after
```
"(* tata *)
```
thus a different approach is probably necessary; I don't think matching comments inside string literals would be a feature. I'm the one to blame for this – sorry for leading you astray!
As mentioned before, the checks for the `*)` characters do the heavy lifting here so the cost of `syntax-ppss` is probably not intolerable in that context.

Reply:
OK, will use `syntax-ppss`.

Reply:
> `"(* tata *)`

If the cursor if after this expression and the default `show-paren-data-function` is called, this result in mismatched parentheses (but does nothing on `"(* tata * )`.  I guess they are still somewhat considered comments somewhere despite being inside a string...

Reply:
Also `*)` at the beginning of a buffer should be considered as mismatched.